node/bft/src/helpers/round_cache.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::{ensure, Result};
+use snarkvm::{
+    console::types::{Address, Field},
+    ledger::committee::Committee,
+    prelude::Network,
+};
+use std::collections::{HashSet, VecDeque};
+
+#[derive(Copy, Clone, Debug)]
+struct AddressWithCoordinate<N: Network> {
```

Comment:
Just riffing but:
```suggestion
struct AddressWithAbscissa <N: Network> {
```
or 
```suggestion
struct AddressWithXCoordinate<N: Network> {
```

Reply:
Not sure these are better than your original suggestion overall.